test/client.jl

```diff
@@ -1,8 +1,24 @@
-using HTTP, Test, JSON
+include("resources/TestRequest.jl")
+using ..TestRequest
 
 @testset "HTTP.Client" begin
 
 status(r) = r.status
+@testset "Custom HTTP Stack" begin
+   @testset "High-level Request" begin
+        custom_stack = insert(stack(), StreamLayer, TestLayer)
+        result = request("GET", "https://httpbin.org/ip", custom_stack=custom_stack)
+
+        @test status(result) == 200
+    end
+
+    @testset "Shorthand Request" begin
+        custom_stack = insert(stack(), StreamLayer, TestLayer)
+        result = HTTP.get("https://httpbin.org/ip", custom_stack=custom_stack)
```

Comment:
On seeing this in use, I still do not like it.

much rather stack was a first argument,
even though it means defining more overloads.

I could be convinced though